Adapter migration mysql

tyrsson opened the pull request:
- Begins moving the MySQL related code to its satellite package.
- Introduces a PdoDriverInterface that extends DriverInterface.

tyrsson committed:
latest changes. Pdo named arguments are broken
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Update StatementInterface prepare signature
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
fix config provider
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Fixes regression of "mapped" parameters thanks to @simon-mundy
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Merge branch '3.0.x' into adapter-migration-mysql

tyrsson committed:
Introduces Driver\Pdo\AbstractPdo
Introduces Laminas\Db\Adapter\Platform\PlatformInterface::getSqlPlatformDecorator()
Improves typing around surface area methods
Adds strict typing to all touched files.

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Missed strict types
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Adding temp debug comments
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Removes MySQL related code that has been migrated to laminas-db-adapter-mysql
Introduces Container\MetadataFactory to replace deprecated Metadata\Source\Factory
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Temp removes ci constraint on which actions trigger workflow runs for the ci action.
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Refactoring around ConnectionInterface, AbstractConnection, AbstractPdoConnection
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

settermjd commented on src/Adapter/Driver/AbstractConnection.php:
```diff
@@ -41,39 +32,24 @@ public function disconnect()
-     */
-    public function getDriverName()
+    /** Get driver name */
+    public function getDriverName(): ?string
```
What about setting the return type of this and others to be a union type, e.g., `string|null` to avoid the deprecation warning in 8.4?

tyrsson replied:
I'll look into it. Didn't realize it was being deprecated. Can you please link me to the information about it. I checked the 8.4 deprecations and unless I just missed it cause I'm on mobile I didn't see it in the list.

Thanks

tyrsson replied:
Ok, so I am pretty sure you are thinking of the implicitly nullable parameter deprecation, which I am aware of which is what confused me.

1. The deprecation in 8.4 does not affect return types
2. It also does not require a union.

Please see:
https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated#change

settermjd replied:
I thought that it did affect return types. My bad.

settermjd replied:
Still, would using a union type not be that much more self-descriptive?

settermjd commented on src/Adapter/Driver/PdoDriverInterface.php:
```diff
@@ -0,0 +1,7 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Db\Adapter\Driver;
+
+interface PdoDriverInterface extends DriverInterface {}
```
Should this be refactored out into the base package, so that all satellites can use it?

tyrsson replied:
It's in the base package. If it were not then it would be in the adapter-mysql repo.

settermjd reviewed (commented):
Just a few comments.

tyrsson committed:
Refactor AdapterAwareInterface and Trait
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
DriverInterface::getDatabasePlatformName() signature change
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Updating StatementInterface typing and \Pdo\Statement typing
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Updating StatementInterface::execute to nullable return type
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Updating \Pdo\Statement::execute to nullable return type
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Refactoring typing
\Pdo\AbstractPdo
\Pdo\Statement
\Driver\DriverInterface
\Adapter\StatementContainerInterface
Introduces DriverAwareInterface, PdoDriverAwareInterface
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

settermjd commented on src/Adapter/AbstractAdapter.php:
```diff
@@ -17,26 +17,8 @@
  * @property Platform\PlatformInterface $platform
  */
-class Adapter implements AdapterInterface, Profiler\ProfilerAwareInterface
+abstract class AbstractAdapter implements AdapterInterface, Profiler\ProfilerAwareInterface
```
Just checking in as to what the motivation was for this change, as it's breaking numerous tests.

simon-mundy replied:
@settermjd Because this used to be a concrete implementation and now each individual adapter will extend this. I'd believe it would be a breaking change. Will probably need to mock an Adapter for testing purposes (e.g a TrustingSql92Adapter)

tyrsson committed:
Fixed type in DriverAwareInterface file/class name
Fixed Interface return types to reflect the Interface being returned
Adds strict typing to all classes / interfaces
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Interface typing
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Fixing formatParameterName parameter typing for $name
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
testing temp fix for formatParameterName
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
updating temp fix to proposed fix, see comment in AbstractPdo::formatParameterName
Typing at DriverInterface definition remains typed as (string $name)
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Aligning for integration test
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Package initialization refactoring
Providing AdapterManager via ServiceManager v4
Updates SM dependency to v4
Updates lock file
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Latest revisions
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

settermjd committed:
Fix up types

The changes in this commit are aimed to, where practical, fix up or
improve the use of types throughout the package. While there are far
more to go, this is a solid continuation of the existing work.

tyrsson committed:
Refactor AdapterServiceFactory to align with AdapterManager implementation
Signed-off-by: Joey Smith <jsmith@webinertia.net>

Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson committed:
Merge branch 'adapter-migration-mysql' of https://github.com/axleus/laminas-db into adapter-migration-mysql

tyrsson committed:
Signed-off-by: Joey Smith <jsmith@webinertia.net>

tyrsson commented:
Closing this PR as it has been superseded by the adapter-migration-mysql-new branch.
